lib/msf/core/evasion.rb

```diff
@@ -101,8 +101,14 @@ def normalize_platform_arch
     # @param [FlaseClass] Payload is not compatible.
     def is_payload_compatible?(name)
       p = framework.payloads[name]
+      return false unless p
```

Comment:
I'm not sure of the origin stories of evasion modules, but this PR is mostly aligning the copy/pasta between the evasion and exploit module implementations

Reply:
Original snippet from: https://github.com/rapid7/metasploit-framework/pull/12085